src/Http/Http.Results/test/ResultsOfTHelperTests.cs

```diff
@@ -0,0 +1,116 @@
+
+        using var remoteHandle = RemoteExecutor.Invoke(static () =>
+        {
+            var methodInfo = typeof(ResultsOfTHelperTests).GetMethod(nameof(PopulateMetadataIfTargetIsIEndpointMetadataProvider_PublicMethod_Called));
+            var endpointBuilder = new TestEndpointBuilder();
+
+            ResultsOfTHelper.PopulateMetadataIfTargetIsIEndpointMetadataProvider<PublicMethodEndpointMetadataProvider>(
+                methodInfo,
+                endpointBuilder);
+
+            Assert.Single(endpointBuilder.Metadata);
+        }, options);
+    }
+
+    [ConditionalTheory]
+    [RemoteExecutionSupported]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void PopulateMetadataIfTargetIsIEndpointMetadataProvider_ExplicitMethod_Called(bool isDynamicCodeSupported)
+    {
+        var options = new RemoteInvokeOptions();
+        options.RuntimeConfigurationOptions.Add("System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported", isDynamicCodeSupported.ToString());
+
+        using var remoteHandle = RemoteExecutor.Invoke(static () =>
+        {
+            var methodInfo = typeof(ResultsOfTHelperTests).GetMethod(nameof(PopulateMetadataIfTargetIsIEndpointMetadataProvider_PublicMethod_Called));
+            var endpointBuilder = new TestEndpointBuilder();
+
+            ResultsOfTHelper.PopulateMetadataIfTargetIsIEndpointMetadataProvider<ExplicitMethodEndpointMetadataProvider>(
+                methodInfo,
+                endpointBuilder);
+
+            Assert.Single(endpointBuilder.Metadata);
+        }, options);
+    }
+
+    [ConditionalTheory]
+    [RemoteExecutionSupported]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void PopulateMetadataIfTargetIsIEndpointMetadataProvider_ExplicitAndPublicMethod_ExplicitCalled(bool isDynamicCodeSupported)
+    {
+        var options = new RemoteInvokeOptions();
+        options.RuntimeConfigurationOptions.Add("System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported", isDynamicCodeSupported.ToString());
+
+        using var remoteHandle = RemoteExecutor.Invoke(static () =>
+        {
+            var methodInfo = typeof(ResultsOfTHelperTests).GetMethod(nameof(PopulateMetadataIfTargetIsIEndpointMetadataProvider_PublicMethod_Called));
+            var endpointBuilder = new TestEndpointBuilder();
+
+            ResultsOfTHelper.PopulateMetadataIfTargetIsIEndpointMetadataProvider<ExplicitAndPublicMethodEndpointMetadataProvider>(
+                methodInfo,
+                endpointBuilder);
+
+            Assert.Single(endpointBuilder.Metadata);
+        }, options);
+    }
+
+    private class TestEndpointBuilder : EndpointBuilder
+    {
+        public override Endpoint Build()
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    private class PublicMethodEndpointMetadataProvider : IEndpointMetadataProvider
```

Comment:
Do we want to add a test for the scenario @vitek-karas mentioned?

```C#

interface IMyEndpointMetadataProvider : IEndpointMetadataProvider
{
    public static void PopulateMetadata(MethodInfo method, EndpointBuilder builder)
    {
        builder.Metadata.Add("Called");
    }
}

private class DefaultInterfaceMethodEndpointMetadataProvider : IMyEndpointMetadataProvider
{
}

ResultsOfTHelper.PopulateMetadataIfTargetIsIEndpointMetadataProvider<DefaultInterfaceMethodEndpointMetadataProvider>(...);
```

Same comment for the static method being in a base class.

(I'm not sure if that compiles or not. but you get the idea).

Reply:
Done. This errors when using reflection. I've added an error message saying that a default interface implementation of PopulateMetadata isn't supported.

Realistically, no one is going to do it. And we will hopefully improve calling static interface methods in the future.